tests/test_converter.py

```diff
@@ -0,0 +1,721 @@
+        qft_qasm = tmpdir.join("qft.qasm")
+
+        with open(qft_qasm, "w") as f:
+            f.write(TestConverterQasm.qft_qasm)
+
+        quantum_circuit = load_qasm_from_file(qft_qasm)
+
+        with pytest.warns(UserWarning) as record:
+            with recorder:
+                quantum_circuit()
+
+        assert len(recorder.queue) == 6
+
+        assert recorder.queue[0].name == 'PauliX'
+        assert recorder.queue[0].params == []
+        assert recorder.queue[0].wires == [0]
+
+        assert recorder.queue[1].name == 'PauliX'
+        assert recorder.queue[1].params == []
+        assert recorder.queue[1].wires == [2]
+
+        assert recorder.queue[2].name == 'Hadamard'
+        assert recorder.queue[2].params == []
+        assert recorder.queue[2].wires == [0]
+
+        assert recorder.queue[3].name == 'Hadamard'
+        assert recorder.queue[3].params == []
+        assert recorder.queue[3].wires == [1]
+
+        assert recorder.queue[4].name == 'Hadamard'
+        assert recorder.queue[4].params == []
+        assert recorder.queue[4].wires == [2]
+
+        assert recorder.queue[5].name == 'Hadamard'
+        assert recorder.queue[5].params == []
+        assert recorder.queue[5].wires == [3]
+
+        assert len(record) == 11
+        # check that the message matches
+        assert record[0].message.args[0] == "pennylane_qiskit.converter The {} instruction is not supported by" \
+                                            " PennyLane, and has not been added to the template."\
+            .format('Barrier')
+        assert record[1].message.args[0] == "pennylane_qiskit.converter The {} instruction is not supported by" \
+                                            " PennyLane, and has not been added to the template."\
+            .format('Cu1Gate')
+        assert record[7].message.args[0] == "pennylane_qiskit.converter The {} instruction is not supported by" \
+                                            " PennyLane, and has not been added to the template."\
+            .format('Measure')
+
+    def test_qasm_(self, recorder):
+        """Tests that a QuantumCircuit object is deserialized from a qasm string."""
+        qasm_string = 'include "qelib1.inc";' \
+                      'qreg q[4];' \
+                      'creg c[4];' \
+                      'x q[0];' \
+                      'x q[2];'\
+                      'measure q -> c;'
+
+        quantum_circuit = load_qasm(qasm_string)
+
+        with pytest.warns(UserWarning) as record:
+            with recorder:
+                quantum_circuit(params={})
+
+        assert len(recorder.queue) == 2
+
+        assert recorder.queue[0].name == 'PauliX'
+        assert recorder.queue[0].params == []
+        assert recorder.queue[0].wires == [0]
+
+        assert recorder.queue[1].name == 'PauliX'
+        assert recorder.queue[1].params == []
+        assert recorder.queue[1].wires == [2]
```

Comment:
Amazing work on the tests! 👍

Reply:
Can we add a couple integration tests? I'm thinking that:
- the Recorder is not used (instead, we use a standard PL Qnode)
- the `assert` checks that the correct output is returned
- also, be good to check when the `load` function is used inline in the QNode (from what I could see, all tests above used `load` before creating the QNode

Reply:
@co9olguy this was a great idea, thank you!

I've added some integration tests and turned out that further checks were needed in `_function` for the case when parameters are of `Variable` type.